lib/item_stream_handler.rb

```diff
@@ -55,6 +55,15 @@ def remove_old_ids(id_hash)
     end
   end
 
+  # Returns true if given item appears to represent a recent checkout. Must:
+  #  - be a Hash
+  #  - have a non-nil status.duedate
+  #  - have an id
+  def item_is_checkout? (item)
+    item.is_a?(Hash) &&
```

Comment:
Do you think it worth logging the error is an item is not a Hash? Does this mean something wrong with the item-updater?

Reply:
The only way for `item` to not be a hash would be for the ItemService (which writes the Item events to the Item stream) to write invalid avro data. I don't know of that ever happening and if it were to happen it would indicate a much bigger problem seen in a lot of other places. So I don't think this is the place to log that kind of thing

Reply:
makes sense.